// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineSentinelOneDestination.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineSentinelOneDestination = void 0;
/**
 * The `sentinel_one` destination sends logs to SentinelOne.
 *
 * **Supported pipeline types:** logs
 */
class ObservabilityPipelineSentinelOneDestination {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineSentinelOneDestination.attributeTypeMap;
    }
}
exports.ObservabilityPipelineSentinelOneDestination = ObservabilityPipelineSentinelOneDestination;
/**
 * @ignore
 */
ObservabilityPipelineSentinelOneDestination.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    inputs: {
        baseName: "inputs",
        type: "Array<string>",
        required: true,
    },
    region: {
        baseName: "region",
        type: "ObservabilityPipelineSentinelOneDestinationRegion",
        required: true,
    },
    type: {
        baseName: "type",
        type: "ObservabilityPipelineSentinelOneDestinationType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
